### PR TITLE
Create timer so user can extend session if necessary

### PR DESCRIPTION
**Pull request by Sgtpluck**
## Description

Addresses #1181. Creates a little javascript timeout to give user a prompt to extend the session right before timeout.

## Additional information
I'm just using the built-in alert bc it was fastest/easiest, but it doesn't look very nice. Looks fairly straightforward to make a little warn template if that seems worth it.

### CODE REVIEW

File: tock/tock/urls.py [hunk 1]
```diff
@@ -39,6 +39,9 @@ def check_if_staff(user):
 
     path('auth/', include('uaa_client.urls')),
 
+    path('session-warning', tock.views.session_warning, name="SessionWarning"),
```
Comment (Sgtpluck) [inline tock/tock/urls.py:42]: this seemed like an okay place to put these endpoints but ... i dunno! is there a better one?

File: tock/tock/templates/base.html [hunk 2]
```diff
@@ -37,6 +37,35 @@
...
+        var request = new XMLHttpRequest();
+        request.open("GET", "/session-extend", true);
+        request.onload = function () {
+          if (request.status === 401) {
+            alert("Your session has expired. Please log back in.");
+            logout();
+          }
+        };
+        request.send();
+        setTimeout(setAlert, timeoutCount - 120 * 1000);
```
Comment (Sgtpluck) [inline tock/tock/templates/base.html:51]: if it succeeds, gotta set a new timer

File: tock/tock/templates/base.html [hunk 3]
```diff
@@ -37,6 +37,35 @@
     ga('set', 'forceSSL', true);
     ga('send', 'pageview');
   </script>
+  <script>
```
Comment (Sgtpluck) [inline tock/tock/templates/base.html:40]: i still don't like having this much logic in a template but trying to move it over to a JS file  led to some annoying race conditions.

File: tock/tock/templates/base.html [hunk 4]
```diff
@@ -37,6 +37,35 @@
...
+      const timeoutCount = "{{x_tock_session_timeout_ms}}";
+
+      function setAlert() {
+        alert(
+          "Your session is about to expire! We will attempt to refresh your session."
+        );
+        extendSession();
+      }
+
+      setTimeout(setAlert, timeoutCount - 120 * 1000);
```
Comment (Jkrzy) [inline tock/tock/templates/base.html:67]: Do we need to modify `timeoutCount` here? I think we're already rendering it on the template in milliseconds.

Comment (Jkrzy) [reply tock/tock/templates/base.html:67]: Ah I see, we're firing the alert 2 minutes prior to the timeout.

File: tock/tock/templates/base.html [hunk 5]
```diff
@@ -37,6 +37,35 @@
...
+
+      function logout() {
+        window.location.reload();
+      }
+
+      const timeoutCount = "{{x_tock_session_timeout_ms}}";
+
+      function setAlert() {
+        alert(
+          "Your session is about to expire! We will attempt to refresh your session."
```
Comment (Jkrzy) [inline tock/tock/templates/base.html:62]: A small words tweak
```suggestion
          "Your session will expire in 2 minutes. Press OK or the enter key to extend your session."
```

**Review by Jkrzy — APPROVED** [review]
Looks good, thanks @Sgtpluck!!!!

Doing this with a USWDS confirmation  modal will give us a bit more control and let us make this even more accessible. Mind writing up a small issue for iterating on this by replacing the `alert` dialog with something where we can better control the title and interpretation by a screenreader.

I'm hopeful there's some existing art we can borrow from another TTS site

**Pull request merged**